src/Orleans.Runtime/Hosting/CoreHostingExtensions.cs

```diff
@@ -49,26 +75,15 @@ public static ISiloHostBuilder Configure(this ISiloHostBuilder builder, string s
         /// <returns>The host builder.</returns>
         public static ISiloHostBuilder Configure(this ISiloHostBuilder builder, string siloName, Action<OptionsBuilder<ClusterOptions>> configureOptions = null)
         {
-            builder.ConfigureServices((context, services) =>
-            {
-                if (!context.Properties.ContainsKey("OrleansServicesAdded"))
+            return builder
+                .ConfigureDefaults()
+                .ConfigureServices((context, services) =>
                 {
-                    services.PostConfigure<SiloOptions>(options => options.SiloName = options.SiloName
-                                           ?? context.HostingEnvironment.ApplicationName
-                                           ?? $"Silo_{Guid.NewGuid().ToString("N").Substring(0, 5)}");
-
-                    services.TryAddSingleton<Silo>();
-                    DefaultSiloServices.AddDefaultServices(context, services);
-
-                    context.Properties.Add("OrleansServicesAdded", true);
-                }
+                    if (!string.IsNullOrEmpty(siloName))
```

Comment:
shouldn't we throw if silo name isn't valid?
Or if null is valid, make it an optional argument to make this explicit?

Reply:
Other extensions intentionally pass null for this value to use the default. I rather not make it an optional argument because with all these overloads of a method simply called `Configure`, we're bound to hit ambiguity